server/src/test/java/com/paypal/selion/grid/servlets/GridStatisticsTest.java

```diff
@@ -0,0 +1,135 @@
+|                                                                                                                     |
+|       http://www.apache.org/licenses/LICENSE-2.0                                                                    |
+|                                                                                                                     |
+|  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed   |
+|  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for  |
+|  the specific language governing permissions and limitations under the License.                                     |
+\*-------------------------------------------------------------------------------------------------------------------*/
+
+package com.paypal.selion.grid.servlets;
+
+import static org.powermock.api.mockito.PowerMockito.spy;
+import static org.powermock.api.support.membermodification.MemberMatcher.method;
+import static org.powermock.api.support.membermodification.MemberModifier.stub;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.ServletOutputStream;
+import javax.servlet.WriteListener;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.openqa.grid.internal.Registry;
+import org.openqa.grid.web.servlet.RegistryBasedServlet;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.testng.PowerMockTestCase;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.paypal.selion.pojos.BrowserInformationCache;
+import com.paypal.selion.pojos.BrowserStatisticsCollection;
+import com.paypal.selion.pojos.BrowserStatisticsCollection.BrowserStatistics;
+
+/**
+ * Tests for GridStatistics servlet
+ */
+@PrepareForTest({ GridStatistics.class, BrowserInformationCache.class, RegistryBasedServlet.class, Registry.class })
+public class GridStatisticsTest extends PowerMockTestCase {
+
+    @BeforeClass
+    public void beforeClass() {
+
+    }
+
+    @Test
+    public void testValidRequest() throws Exception {
+        HttpServletRequest request = PowerMockito.mock(HttpServletRequest.class);
+        PowerMockito.when(request.getHeader("Accept")).thenReturn("*/*");
+        HttpServletResponse response = PowerMockito.mock(HttpServletResponse.class);
+        ServletOutputStream servletOutputStream = new ServletOutputStream() {
+            ByteArrayOutputStream bos = new ByteArrayOutputStream();
+
+            @Override
+            public void write(int b) throws IOException {
+                bos.write(b);
+            }
+
+            @Override
+            public void setWriteListener(WriteListener arg0) {
```

Comment:
can be ignored .. code will go away once `MockHttpServletResponse` is used instead.